x/coinswap/client/cli/tx.go

```diff
@@ -0,0 +1,182 @@
+package cli
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/context"
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/version"
+	"github.com/cosmos/cosmos-sdk/x/auth"
+	"github.com/cosmos/cosmos-sdk/x/auth/client/utils"
+	"github.com/cosmos/cosmos-sdk/x/coinswap/internal/types"
+)
+
+// GetTxCmd returns the transaction commands for this module
+func GetTxCmd(storeKey string, cdc *codec.Codec) *cobra.Command {
+	coinswapTxCmd := &cobra.Command{
+		Use:                        types.ModuleName,
+		Short:                      "Coinswap transaction subcommands",
+		DisableFlagParsing:         true,
+		SuggestionsMinimumDistance: 2,
+		RunE:                       client.ValidateCmd,
+	}
+
+	coinswapTxCmd.AddCommand(client.PostCommands(
+		GetCmdAddLiquidity(cdc),
+		GetCmdRemoveLiquidity(cdc))...)
+
+	return coinswapTxCmd
+}
+
+// GetCmdAddLiquidity implements the add liquidity command handler
+func GetCmdAddLiquidity(cdc *codec.Codec) *cobra.Command {
+	return &cobra.Command{
+		Use:   "add-liquidity [deposit] [deposit-amount] [min-reward] [deadline]",
+		Args:  cobra.ExactArgs(4),
+		Short: "Add liquidity in the reserve pool",
+		Long: strings.TrimSpace(
+			fmt.Sprintf(`Add liquidity in the reserve pool for a trading pair.
+			
+Example:
+$ %s tx coinswap add-liquidity dai 1000atom 1000 2 cosmosvaloper1l2rsakp388kuv9k8qzq6lrm9taddae7fpx59wm --from mykey
+`,
+				version.ClientName,
+			),
+		),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			txBldr := auth.NewTxBuilderFromCLI().WithTxEncoder(auth.DefaultTxEncoder(cdc))
+			cliCtx := context.NewCLIContext().WithCodec(cdc)
+
+			deposit, err := sdk.ParseCoin(args[0])
+			if err != nil {
+				return err
+			}
+
+			depositAmount, ok := sdk.NewIntFromString(args[1])
+			if !ok {
+				return fmt.Errorf(types.ErrNotPositive(types.DefaultCodespace, "deposit amount provided is not positive").Error())
+			}
+
+			minReward, ok := sdk.NewIntFromString(args[2])
+			if !ok {
+				return fmt.Errorf(types.ErrNotPositive(types.DefaultCodespace, "minimum liquidity is not positive").Error())
+			}
+
+			deadline, err := time.Parse(time.RFC3339, args[3])
+			if err != nil {
+				return err
+			}
+
+			senderAddr := cliCtx.GetFromAddress()
+
+			msg := types.NewMsgAddLiquidity(deposit, depositAmount, minReward, deadline, senderAddr)
+			return utils.GenerateOrBroadcastMsgs(cliCtx, txBldr, []sdk.Msg{msg})
+		},
+	}
+}
+
+// GetCmdRemoveLiquidity implements the remove liquidity command handler
+func GetCmdRemoveLiquidity(cdc *codec.Codec) *cobra.Command {
+	return &cobra.Command{
+		Use:   "remove-liquidity [withdraw] [withdraw-amount] [min-native] [deadline]",
```

Comment:
same with `withdraw` and `withdraw-amount`. Can these be renamed or are they the same?

Reply:
Hey Fede, I agree they seem confusing. However, both are not the same. Deposit denotes the non-native denom, while the deposit amount is the amount of native denom that gets added. Here, the non-native denom and native denom together form a trading pair. Similarly for withdraw and withdraw-amount. Will change these names to denote the same.

Reply:
Hey Fede, I just realized that for the remove liquidity, the above doesn't hold true. For remove liquidity, they follow something known as `Pool Tokens`. Whenever someone adds liquidity, they get UNI tokens back. The same UNI tokens are provided as an input when liquidity needs to be removed. Thus withdraw is the non-native denom, while withdraw amount is the number of UNI tokens. The output of the same is the amount of each denom that together form a trading pair. For instance: if the non-native denom is DAI, and I provide 50 UNI tokens as an input, the output as per Uniswap is X ETH + Y DAI. For more clarification check, https://uniswap.exchange/remove-liquidity

Reply:
Thus I'm going to change `withdraw-amount` to `pool-tokens`. As for the case of add liquidity, what I said earlier stands true i.e. the `deposit-amount` is the amount of native denom that gets added to the pool. I'm going to change `deposit amount` to `deposit`